### ISSUE TITLE
Sending transactions to yourself reduces your balance temporarily 

### ISSUE THREAD

**Issue by KeeJef** [loki-project/loki#644]
**Describe the bug**
When sending a transaction where the origin address is the same as the destination address (Staking or Sweep All) The loki-wallet-cli wallet shows a pending transaction which deducts funds from the balance (Locked and unlocked). 

Pending transactions that you send to yourself should be rendered as transactions for a 0 amount (how they show once one confirmation has arrived) 

Right now when staking or using sweep all a user can panic as it appears that they have spent their balance and they don't see that the transaction is actually not going anywhere. Once they receive a single confirmation they see the true balance. 

This issue should also reference #398 , which describes a new field called pending balance, which could also be a solution to this.

**To Reproduce**

Send a transaction from the origin wallet to the origin wallet, you should see your balance reduce until a single confirmation is processed 

**Screenshots or Log files**
If applicable, add screenshots or Log files to help explain your problem.

**Desktop (please complete the following information):**
 - OS: [e.g. Ubuntu 16.04, Windows 10] ALL
 - Version of Wallet or Git commit hash ALL

**Comment by Fez29** [comment]
I like the pending balance/incoming funds field to inform the user that their funds are safe and are on their way "back" to their wallet, seems trivial to users that have performed these transactions before but can cause alarm for user that are not familiar with these transactions.

**Comment by CryptoFirefly** [comment]
The concept of unspent outputs may not be fully understood by all users so we should simpilfy it and make it error proof, like doing a sweep_all to a wrong address or some other address. 

Enhancements i suggest are
1. The sweep_all command should not require the wallet parameter. As everyone using sweep_all is actually trying to compress the value into one or few unspent outputs rather than sending it to someone.

Instead if someone wants to send to another wallet they can use the usual command to transfer funds. And if unable to send or stake, do sweep_all first and as a next step do the transfer. This will make it safer.

*Asking people to send funds to themselves (which we know does the same) isn't the best way to manage this requirement. A wrong copy paste and someone could loose all his funds. Instead a safe button to just sweep_all does the job safely with whatever fee needed. 

2. Make available the sweep_all feature in GUI / Mobile wallets, again without wallet parameter so using that there is never any accidental move of funds outside of the owners wallet. It will only move unspent outputs into a one big single unspent output. 

3. We name Sweep_all to a more user friendly name on GUI / Mobile wallets which practically does what the sweep_all does, as its a needed feature for a currency like Loki which stores value in unspent outputs. 

With some detailed explanation in a tool-tip or help location, which explains the process in easy lay-man language. 

Suggestions for the Button / Feature: 
a. Consolidate Loki Holdings
b. Bunch-up Loki
c. Bundle-up Loki
others am sure our community can suggest.

**Comment by jagerman** [comment]
> 2. Make available the sweep_all feature in GUI / Mobile wallets, again without wallet parameter so using that there is never any accidental move of funds outside of the owners wallet. It will only move unspent outputs into a one big single unspent output.

The GUI wallet supports this already -- there is an "all" button (or something like that) in the transfer field, and if you selected it you get a sweep (i.e. sends all minus whatever the fee is).

**Comment by CryptoFirefly** [comment]
> > 1. Make available the sweep_all feature in GUI / Mobile wallets, again without wallet parameter so using that there is never any accidental move of funds outside of the owners wallet. It will only move unspent outputs into a one big single unspent output.
> 
> The GUI wallet supports this already -- there is an "all" button (or something like that) in the transfer field, and if you selected it you get a sweep (i.e. sends all minus whatever the fee is).

True, what i actually meant was, say if someone wants to stake and is getting an issue. In that case we recommend send all to ourselves. But to avoid the address parameter (potential risk of sending to a wrong address / someone else), what if a button allows to pack all small lose change txn's (unspend balance) into big chunks. Without having to worry about copying the public address or using the Send all. 

But i see you point, probably the SN operators / contributors are limited in the big picture of users.

**Comment by SomethingGettingWrong** [comment]
![image](https://user-images.githubusercontent.com/36722911/60049690-72fcb100-9694-11e9-8e18-b8dcb206be1f.png)

lol I saw this and thought I put in a wrong.. freaking address.. cutting off a letter sending to an address that didn't exist.. and it took 16 minutes to go from pending.. OMG lol.. when it confirmed turned to 0 and I stopped swearing knowing I had sent it to myself.

**Comment by SomethingGettingWrong** [comment]
Would be nice if it popped up when you send to the same address. Are you sure you want to send to yourself? This is only required when sweeping are staking!  It would know to prompt from the fact it sent to itself. and then it immedietly show 0 in pending

**Comment by SomethingGettingWrong** [comment]
> I like the pending balance/incoming funds field to inform the user that their funds are safe and are on their way "back" to their wallet, seems trivial to users that have performed these transactions before but can cause alarm for user that are not familiar with these transactions.

well when it happens for 17 minutes... and previously only took 1 minute tops.. it definteily is casue for alarm. lol
